src/main/java/bots/FriendlyBot.java

```diff
@@ -0,0 +1,166 @@
+                            };
+                            break;
+                        case "deadline":
+                            try {
+                                checkForDescriptionAndDeadline(line);
+                                int deadlineBy = line.indexOf("/by") + 4;
+                                String by = line.substring(deadlineBy);
+                                checkForDeadlineDescription(line, deadlineBy);
+                                tasks[taskCount] = new FriendlyBotDeadline(line, by);
+                                tasks[taskCount].description = line.substring(9, deadlineBy - 4);
+                                taskCount = displayNewTask(taskCount, tasks[taskCount]);
+                                break;
+                            } catch (NoDescriptionAndDeadlineException ndde) {
+                                System.out.println("Oops! Are you missing a description AND a deadline?? 😱");
+                            } catch (StringIndexOutOfBoundsException e) {
+                                System.out.println("Oops! Are you missing a deadline? 🧐 (use '/by'!)");
+                            } catch (NoDeadlineDescriptionException nde) {
+                                System.out.println("Oops! Are you missing a description for your deadline? 🧐");
+                            }
+                            break;
+                        case "event":
+                            try {
+                                checkForDescriptionAndEventAt(line);
+                                int eventAt = line.indexOf("/at") + 4;
+                                String at = line.substring(eventAt);
+                                checkForEventDescription(line, eventAt);
+                                tasks[taskCount] = new FriendlyBotEvent(line, at);
+                                tasks[taskCount].description = line.substring(6, eventAt - 4);
+                                taskCount = displayNewTask(taskCount, tasks[taskCount]);
+                                break;
+                            } catch (NoDescriptionAndEventAtException ndaea) {
+                                System.out.println("Oops! Are you missing a description AND a time/place for your " +
+                                        "event? 😱");
+                            } catch (StringIndexOutOfBoundsException e) {
+                                System.out.println("Oops! Are you missing a time/place for your event? 🧐 (use '/at'!)");
+                            } catch (NoEventDescriptionException nde) {
+                                System.out.println("Oops! Are you missing a description for your event? 🧐");
+                            }
+                            break;
+                        default:
+                            invalidCommand();
+                            break;
+                    }
+                }
+                System.out.println();
+            }
+        }
+    }
```

Comment:
This is too long. You should try to refactor this portion of the code.

Reply:
Hi Joel, I've refactored my whole Ui class! :)